src/Validation/Constraint/IssuedBy.php

```diff
@@ -0,0 +1,47 @@
+<?php
+/**
+ * This file is part of Lcobucci\JWT, a simple library to handle JWT and JWS
+ *
+ * @license http://opensource.org/licenses/BSD-3-Clause BSD-3-Clause
+ */
+
+declare(strict_types=1);
+
+namespace Lcobucci\JWT\Validation\Constraint;
+
+use DateTimeInterface;
+use Lcobucci\JWT\Validation\ConstraintViolationException;
+use Lcobucci\JWT\Token;
+use Lcobucci\JWT\Validation\Constraint;
+
+/**
+ * @author Luís Otávio Cobucci Oblonczyk <lcobucci@gmail.com>
+ * @since 4.0.0
+ */
+final class IssuedBy implements Constraint
+{
+    /**
+     * @var array
+     */
+    private $issuers;
+
+    public function __construct(string ...$issuers)
+    {
+        $this->issuers = $issuers;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function assert(Token $token)
+    {
+        $claimSet = $token->claims();
+
+        if (!$claimSet->has('iss')
+            || !in_array($token->claims()->get('iss'), $this->issuers, true)) {
```

Comment:
> There's no insurance that `'iss'` is an array here

Issuer is always a `stringOrUri`, so `'iss' is always a string.

The ideia of this constraint is to assert that the token was issued by issuer "a", "b" or "c".